packages/components/pie-webc-testing/src/helpers/page-object/base-page.ts

```diff
@@ -1,10 +1,77 @@
 import { type Page } from '@playwright/test';
+import { buildUrl } from './storybook-extensions';
+
+declare global {
+    interface Window { __eventsArray: Array<string>; }
+}
 
 export class BasePage {
     readonly page: Page;
+    componentName: string;
+    componentTag: string;
+    path: string;
+    args: string;
 
-    constructor (page: Page) {
+    constructor (page: Page, componentName: string, componentTag = 'data-test-id') {
         this.page = page;
+        this.componentName = componentName;
+        this.componentTag = componentTag;
+        this.path = '';
+        this.args = '';
+    }
+
+    async load (queries: Record<string, unknown> = {}) {
+        const pageUrl = buildUrl(this.componentName, this.composePath(queries), this.args);
+        await this.open(pageUrl);
+    }
+
+    async open (url: string) {
+        await this.page.goto(url, { waitUntil: 'networkidle' });
+        return this;
+    }
+
+    composePath (queries: Record<string, unknown>) {
```

Comment:
Might be worth documenting this method.

Reply:
Yup. I'll add it in as a follow up PR.

Reply:
I think an example input -> output would be valuable